### PR TITLE
feat: Add file path handling to PostFileModificationDateJob and change modification time in locale

### PR DESCRIPTION
**Pull request by RaphaelOstier**
This pull request enhances the `PostFileModificationDateJob` class by improving how file modification dates are handled and tested. The changes introduce more robust handling of file paths and server responses, as well as improved test coverage for edge cases involving future timestamps.

**Enhancements to file modification date handling:**

* The `PostFileModificationDateJob` constructor now takes an additional `filePath` parameter, allowing the job to update local file dates after a successful server response. (`src/libsyncengine/jobs/network/kDrive_API/postfilemodificationdatejob.cpp`, `src/libsyncengine/jobs/network/kDrive_API/postfilemodificationdatejob.h`, `test/libsyncengine/jobs/network/testnetworkjobs.cpp`) [[1]](diffhunk://#diff-36ffd882c8708ac2ed431d2a944b43ae38cf019a183d350497410d99627d6582R21-R35) [[2]](diffhunk://#diff-2c252d2f5464ee88faad59f47c5fc3e3ade9c4f445c178d66e70c9c11c8baa68L27-R41) [[3]](diffhunk://#diff-a44011b4b2142730cdb600b754fc9df3f32479959030cd13cf167fda5bfc52ebL1922-R1921)
* Added a `runJob()` override to update the local file's modification date using `IoHelper::setFileDates`, and a `handleResponse()` override to extract the updated timestamp from the server response. (`src/libsyncengine/jobs/network/kDrive_API/postfilemodificationdatejob.cpp`, `src/libsyncengine/jobs/network/kDrive_API/postfilemodificationdatejob.h`) [[1]](diffhunk://#diff-36ffd882c8708ac2ed431d2a944b43ae38cf019a183d350497410d99627d6582R57-R83) [[2]](diffhunk://#diff-2c252d2f5464ee88faad59f47c5fc3e3ade9c4f445c178d66e70c9c11c8baa68L27-R41)

**Improvements to testing:**

* The test for posting file modification dates now passes the file path to the job and adds a new test case to verify server-side capping of modification dates set far in the future. (`test/libsyncengine/jobs/network/testnetworkjobs.cpp`) [[1]](diffhunk://#diff-a44011b4b2142730cdb600b754fc9df3f32479959030cd13cf167fda5bfc52ebR1934-R1956) [[2]](diffhunk://#diff-a44011b4b2142730cdb600b754fc9df3f32479959030cd13cf167fda5bfc52ebL1922-R1921)
* Removed a redundant test case for future dates, replaced with a more thorough check for server capping behavior. (`test/libsyncengine/jobs/network/testnetworkjobs.cpp`)

These changes make file modification date handling more accurate and ensure that edge cases are properly tested.

### CODE REVIEW

File: test/libsyncengine/jobs/network/testnetworkjobs.cpp [hunk 1]
```diff
@@ -1931,6 +1930,34 @@ void TestNetworkJobs::testPostFileModificationDate() {
...
+        CPPUNIT_ASSERT(testJob.runSynchronously());
+
+        GetFileInfoJob verifyJob(_driveDbId, nodeId);
+        CPPUNIT_ASSERT_EQUAL(ExitCode::Ok, verifyJob.runSynchronously().code());
+        Poco::JSON::Object::Ptr verifyDataObj = verifyJob.jsonRes()->getObject(dataKey);
+        CPPUNIT_ASSERT(verifyDataObj);
+        SyncTime verifyLastModifiedAt = 0;
+        CPPUNIT_ASSERT(JsonParserUtility::extractValue(verifyDataObj, lastModifiedAtKey, verifyLastModifiedAt, false));
+        CPPUNIT_ASSERT_LESS(invalidFutureModificationDate, verifyLastModifiedAt);
+        CPPUNIT_ASSERT_EQUAL(testJob.lastModifiedAt(), verifyLastModifiedAt);
```
Comment (herve-er) [inline test/libsyncengine/jobs/network/testnetworkjobs.cpp]:
```suggestion
        CPPUNIT_ASSERT_EQUAL(testJob.lastModifiedAt(), verifyJob->modificationJob());
```

And remove the extraction part as it is already done by the job

File: test/libsyncengine/jobs/network/testnetworkjobs.cpp [hunk 2]
```diff
@@ -1931,6 +1930,34 @@ void TestNetworkJobs::testPostFileModificationDate() {
...
+        PostFileModificationDateJob testJob(_driveDbId, nodeId, invalidFutureModificationDate);
+        CPPUNIT_ASSERT(testJob.runSynchronously());
+
+        GetFileInfoJob verifyJob(_driveDbId, nodeId);
+        CPPUNIT_ASSERT_EQUAL(ExitCode::Ok, verifyJob.runSynchronously().code());
+        Poco::JSON::Object::Ptr verifyDataObj = verifyJob.jsonRes()->getObject(dataKey);
+        CPPUNIT_ASSERT(verifyDataObj);
+        SyncTime verifyLastModifiedAt = 0;
+        CPPUNIT_ASSERT(JsonParserUtility::extractValue(verifyDataObj, lastModifiedAtKey, verifyLastModifiedAt, false));
+        CPPUNIT_ASSERT_LESS(invalidFutureModificationDate, verifyLastModifiedAt);
```
Comment (herve-er) [inline test/libsyncengine/jobs/network/testnetworkjobs.cpp]: inverted

**Review by herve-er — APPROVED** [review]
